waku/v2/protocol/relay/waku_relay.go

```diff
@@ -495,3 +527,43 @@ func (w *WakuRelay) Params() pubsub.GossipSubParams {
 func (w *WakuRelay) Events() event.Bus {
 	return w.events
 }
+
+func (w *WakuRelay) addPeerTopicEventListener(topic *pubsub.Topic) (*pubsub.TopicEventHandler, error) {
+	handler, err := topic.EventHandler()
+	if err != nil {
+		return nil, err
+	}
+	w.WaitGroup().Add(1)
+	go w.topicEventPoll(topic.String(), handler)
+	return handler, nil
+}
+
+func (w *WakuRelay) topicEventPoll(topic string, handler *pubsub.TopicEventHandler) {
+	defer w.WaitGroup().Done()
+	for {
+		evt, err := handler.NextPeerEvent(w.Context())
+		if evt.Peer == "" { //Empty peerEvent is returned when context passed in done.
```

Comment:
go-libp2p has the Validate() method which verifies if the peer is nil or not.
```suggestion
		if evt.Peer.Validate() != nil { //Empty peerEvent is returned when context passed in done.
```
Also, this verification should probably go after the `if err != nil` on line 548, since we usually check for errors first before checking for the value returned by a function